lib/virt.py

```diff
@@ -676,12 +700,16 @@ def snapshotted(self):
         """
         Create a snapshot, restore the guest, ready it for communication.
         """
-        if not self.can_be_snapshotted():
-            raise RuntimeError(f"guest {self.name} not ready for snapshotting")
+        if not self.is_installed():
+            raise RuntimeError(f"guest {self.name} not installed or installed with different tag")
+        if not self.snapshot_ready:
+            raise RuntimeError(
+                f"guest {self.name} not prepared for snapshotting, "
+                "prepare_for_snapshot() needs to be used first",
+            )
         self._restore_snapshotted()
-        if not self.ipaddr:
-            self.ipaddr = wait_for_ifaddr(self.name)
-            wait_for_ssh(self.ipaddr)
+        self.ipaddr = wait_for_ifaddr(self.name)
+        wait_for_ssh(self.ipaddr)
```

Comment:
I believe this was an important optimization for fast snapshots - with week+ lease time, it's unlikely the VM changes IP address during test runtime, and waiting for `virsh domifaddr` takes extra time on every restore.